deviations/views.py

```diff
@@ -44,39 +94,42 @@ def form_valid(self, form):
             return super().form_valid(form)
 
         without_late_penalty = form.cleaned_data["without_late_penalty"]
+        already_have_deviation = []
         for profile in form.cleaned_data["submitter"]:
+            deviation_exercises = form.cleaned_data["exercise"]
             for module in form.cleaned_data["module"]:
-                exercises = BaseExercise.objects.filter(
-                    course_module = module
-                )
-                for exercise in exercises:
-                    if new_date:
-                        minutes = exercise.delta_in_minutes_from_closing_to_date(
-                                new_date)
-                    self.add_deviation(
-                        exercise, profile, minutes, without_late_penalty)
-
-            for exercise in form.cleaned_data["exercise"]:
+                exercises = BaseExercise.objects.filter(course_module=module)
```

Comment:
Does this query the exercises from the database for each user? It is unnecessary to query the same exercises multiple times.